3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -118,31 +70,28 @@ function add_member() {
   local replset_addr
   replset_addr="${MONGODB_REPLICA_NAME}/${endpoints}"
 
-  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${script}" --quiet; then
+  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "while (!rs.add('${host}').ok) { sleep(100); }" --quiet; then
```

Comment:
I understand, this is difference: petset vs. replset. I think DNS does not brings no advantage when addresing petset (from mongodb clients).

IP addresses are not stored anywhere. The are got by the same way and in same time as DNS names (by `dig` command from service), so they points to the same containers while addressing mongodb replicaset. Only when using DNS names, another DNS query have to be done (immediately after getting DNS names from service) to get IP addresses.

However it is not important :-)